perception/map_based_prediction/src/map_based_prediction_node.cpp

```diff
@@ -585,6 +585,54 @@ bool hasPotentialToReach(
   return false;
 }
 
+template <typename T>
+std::unordered_set<std::string> removeOldObjectsHistory(
+  const double current_time, const double buffer_time,
+  std::unordered_map<std::string, std::deque<T>> & target_objects)
+{
+  std::unordered_set<std::string> invalid_object_id;
+  for (auto iter = target_objects.begin(); iter != target_objects.end(); ++iter) {
+    const std::string object_id = iter->first;
+    std::deque<T> & object_data = iter->second;
+
+    // If object data is empty, we are going to delete the buffer for the obstacle
+    if (object_data.empty()) {
+      invalid_object_id.insert(object_id);
+      continue;
+    }
+
+    const double latest_object_time = rclcpp::Time(object_data.back().header.stamp).seconds();
+
+    // Delete Old Objects
+    if (current_time - latest_object_time > buffer_time) {
+      invalid_object_id.insert(object_id);
+      continue;
+    }
+
+    // Delete old information
+    while (!object_data.empty()) {
+      const double post_object_time = rclcpp::Time(object_data.front().header.stamp).seconds();
+      if (current_time - post_object_time > buffer_time) {
+        // Delete Old Position
+        object_data.pop_front();
+      } else {
+        break;
+      }
+    }
```

Comment:
I mean something like this: which is more readable and will not significantly affect performance
```suggestion
    std::remove_if(object_data.begin(),object_data.end(),[&](const auto & obj){
    const double post_object_time = rclcpp::Time(obj.header.stamp).seconds();
    return  (current_time - post_object_time > buffer_time);
    });
```
You could also do this: 
```suggestion
    while (!object_data.empty()) {
      const double post_object_time = rclcpp::Time(object_data.front().header.stamp).seconds();
      if (current_time - post_object_time <= buffer_time) {
        break;
      }
      object_data.pop_front();
    }
```
But the first suggestion is better imo.

Reply:
Thank you! But as I understand the first suggestion does not have early exit, it will keep iterating through the list getting bool value for all of the objects. I will go with the second solution.

Reply:
@dkoldaev yes, it does not have but the processing cost is low, I believe the first option is more readable and the performance cost is not significant.